# Case Study -2

In [2]:
import numpy as np
import pandas as pd

In [3]:
file=pd.read_csv("StudentsPerformance.csv")
data=pd.DataFrame(file)


In [4]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

## 1. Find out how many males and females participated in the test.

In [5]:
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

## 2. What do you think about the students parental level of education?

In [6]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Looking at the data we can observe that most of the parents have a college degree. There are also parents with masters degree
but their number is low compared to the others.

## 3. Who scores the most on average for math, reading and writing based on:

### a) Gender

In [7]:
data1=data.groupby('gender')['math score'].mean()

In [8]:
data1

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64

Males scored more in math compared to females.

In [9]:
data1=data.groupby('gender')['reading score'].mean()

In [10]:
data1

gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64

Females score more in reading compared to males.

In [11]:
data1=data.groupby('gender')['writing score'].mean()

In [12]:
data1

gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64

Females also scored more in writing compared to males.

### b) Test preparation course

In [13]:
data1=data.groupby('test preparation course')['math score'].mean()

In [14]:
data1

test preparation course
completed    69.96648
none         63.78972
Name: math score, dtype: float64

Those who completed the test preparation course scored more in math.

In [15]:
data1=data.groupby('test preparation course')['reading score'].mean()

In [16]:
data1

test preparation course
completed    74.175978
none         66.417445
Name: reading score, dtype: float64

Those who completed the test preparation course also scored more in reading than the others who didn't.

In [17]:
data1=data.groupby('test preparation course')['writing score'].mean()

In [18]:
data1

test preparation course
completed    74.684358
none         64.457944
Name: writing score, dtype: float64

Again those who completed the test preparation course also scored more in reading than the others who didn't.

## 4. What do you think about the scoring variation for math, reading and writing based on:

### a) Gender

In [19]:
data1=data.groupby('gender')['math score'].std()

In [20]:
data1

gender
female    16.029928
male      14.556411
Name: math score, dtype: float64

In [21]:
data1=data.groupby('gender')['reading score'].std()

In [22]:
data1

gender
female    14.411018
male      14.149594
Name: reading score, dtype: float64

In [23]:
data1=data.groupby('gender')['writing score'].std()

In [24]:
data1

gender
female    14.844842
male      14.227225
Name: writing score, dtype: float64

The males were consistent with their score in math, reading and writing. There was a slight variation with the scores
of females. It dropped in reading and writing scores.

### b) Test preparation course

In [25]:
data1=data.groupby('test preparation course')['math score'].std()

In [26]:
data1

test preparation course
completed    14.521847
none         15.705689
Name: math score, dtype: float64

In [27]:
data1=data.groupby('test preparation course')['reading score'].std()

In [28]:
data1

test preparation course
completed    13.537572
none         14.608896
Name: reading score, dtype: float64

In [29]:
data1=data.groupby('test preparation course')['writing score'].std()

In [30]:
data1

test preparation course
completed    13.236412
none         15.041667
Name: writing score, dtype: float64

Here those who completed the Test preparation course came out with a better consistency in scores compared to those who didnt 
complete the course.

## 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.


In [31]:
top250=data.sort_values(by='math score',ascending=False).head(n=250)


In [32]:
top250

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]

There are a total of 1000 students(518 females and 482 males). 25% means that 250 of the 1000 students who got the highest score in math should get the bonus points. So by sorting in descending order on the basis of math score uptil 250 students, we get the top 250 who are eligible for the bonus points.